app/models/job_proxy_dispatcher.rb

```diff
@@ -144,11 +174,12 @@ def assign_proxy_to_job(proxy, job)
     end
   end
 
-  def pending_jobs
+  def pending_jobs(containers = false)
     @zone = MiqServer.my_zone
     Job.order(:id)
       .where(:state           => "waiting_to_start")
       .where(:dispatch_status => "pending")
+      .where("target_class #{containers ? "=" : "!="} ?", ContainerImage.name)
```

Comment:
Sorry, yes, that's what I meant.  If you wanted to be careful, you could do this

```ruby
def pending_jobs(target_class = VmOrTemplate)
  class_name = target_class.base_class.name
  ...
  .where(:target_class => class_name)
end
```

Then, call that with `pending_jobs(ContainerImage)` and leave the vm scanning stuff alone.

And follow the same pattern with `active_scans`.

What do you think?  This would eliminate the conditional and make the parameter a little more obvious.